Login email or username qa

lenacom opened the pull request.

lenacom committed:
Login by email or username

lenacom committed:
fixed check

dmitry-viskov commented on common/djangoapps/student/tests/test_login.py:
```diff
@@ -87,6 +87,14 @@ def test_login_fail_no_user_exists(self):
 
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_LOGIN_BY_EMAIL_OR_USERNAME': True})
+    def test_login_success_by_username(self):
+        response, mock_audit_log = \
```
```
        response, mock_audit_log = self._login_response('test', 'test_password',
                                                        patched_audit_log='student.models.AUDIT_LOG')
```

this looks more elegantly than

```
        response, mock_audit_log = \
            self._login_response('test', 'test_password',
                                 patched_audit_log='student.models.AUDIT_LOG')
```

lenacom replied:
I agree but the 2d line is > 80 chars

dmitry-viskov commented on openedx/core/djangoapps/user_api/views.py:
```diff
@@ -65,24 +65,30 @@ def get(self, request):
         # Translators: These instructions appear on the login form, immediately
         # below a field meant to hold the user's email address.
-        email_instructions = _("The email address you used to register with {platform_name}").format(
+        email_instructions = _("The email address")
```
`_( )` is `gettext` literals. So all such string (after parse this code with the special command) will be saved into special `po` files. I think because of this you can't split this line into separate words. The better way to create `if-else` expression with full sentence in case of 'True`and with the full sentence in case of 'False`

Commit:
fixed comment

Commit:
fixed comment

Commit:
fixed comment (+4 squashed commit)

Squashed commit:

[02035e6] fixed comment

[c419dbe] fixed comment

[b40ae9c] fixed comment

[81d22ff] fixed check

Commit:
fixed comment

Commit:
fixed comment

Commit:
Merge branch 'login-email-or-username-qa' of https://github.com/CredoReference/edx-platform into login-email-or-username-qa
